### ISSUE TITLE
Wmean

### ISSUE THREAD

**Issue by ldutoit** [colin-olito/XvAutosomeInversions#6]
functions-WFSims-Autosomal.R: It seems to me that W.mean has two different meanings if it is in save trajerctories or not within the function autoInvFwdSim ( line 273 or 333), is that right ?

**Comment by colin-olito** [comment]
Hmm... 

Again, I don't have time to take a close look tonight.... but as I recall... no. They shouldn't be any different. It's just that if you save the trajectories, you get a time series of mean fitness for each generation of each simulation as output. does that make sense?

**Comment by ldutoit** [comment]
cool!

I am still trying to understand.  But maybe it is not that important.

Here is some pseudocode
In your autosomal case, 

``
if savetrajectories:
Wbar   <-  sum(O*W)
W.mean[gen+1]     <-  Wbar
W.mean      =  W.mean[1:gen-1]

if !savetrajectories:

Wbar   <-  sum(O*W)
W.mean     <-  Wbar
 
W.mean     =  sum(W.mean)/length(W.mean) # in the res storage
``

This  sum(W.mean)/length(W.mean) is confusing me, is not leangth of W.mean always one?

Thank you again for the collaboration that is really a supervision :)

**Comment by colin-olito** [comment]
Hi Ludo,

You are exactly right - the .../length(W.mean) part doesn't make sense because Wbar is overwritten each generation. So basically, W.mean is just going to be the last value of Wbar for that simulation... which won't change if it is divided by length(W.mean), because this equals 1!

I suspect this was a mistake I made as I introduced the saveTrajectories vs. !saveTrajectories options... dividing by the length would make sense if you were storing the whole vector of Wbar values, and wanted to calculate an overall mean for the simulation. However, after some fiddling, I decided not to do this in the saveTrajectories version because it would be an unnecessary calculation inside a loop that could easily be done afterward with the output using ddply(). I suspect that I forgot to change this when I copy/pasted and made the !saveTrajectories version of the code.
